internal/server/grpc/grpc.go

```diff
@@ -47,30 +51,75 @@ func New(ctx context.Context, cancel context.CancelFunc, name string, config ser
 
 func (s *Server) Start() error {
 	errCh := make(chan error)
+	grpcServerOptions := []grpc.ServerOption{
+		grpc.UnaryInterceptor(otelgrpc.UnaryServerInterceptor()),
+	}
 
 	listener, err := net.Listen("tcp", s.Address)
 	if err != nil {
 		return fmt.Errorf("failed to listen on port %s: %w", s.Address, err)
 	}
+	creds := grpc.Creds(insecure.NewCredentials())
 
 	switch {
 	case s.Config.CertFile != "" || s.Config.KeyFile != "":
```

Comment:
```suggestion
	case s.Config.CertFile != "" && s.Config.KeyFile != "":
```

Reply:
@rodneyosodo 
Since there is no variable to check weather user is opted from TLS , 
So if any one of the file is given then assumes user is opted for TLS.

If user fails to provide two file, Instead provided one file, Then program will run without TLS 
So in order to notify the uesr, if one file is provided it will lead to error , So by user will get notified

Reply:
Understood